### DISCUSSION TITLE
CFWheels v2.4.0 Release

### DISCUSSION THREAD

**Discussion by bpamiri** [wheels-dev/wheels#1174] [Announcements]
Hello everyone, we just release the latest version of the framework. This release contains some minor enhancements and bug fixes and welcomes a couple of first time contributors to the project. Read the full blog post here https://cfwheels.org/blog/cfwheels-v2-4-0-released/.

**Comment by edthetechie** [comment]
I'd just like to say a big thank you to everyone who worked on the new release. I don't know exactly what you did but a whole section of my application just speeded up. I'm talking about a complex user profile form going from taking 4-5 seconds to less than a second. The page talks to a massive dataset with multiple linked tables and I never figured out why it was so slow. That problem went away when I put 2.4.0 on and I've got compliments flooding in from the client. THANK YOU!

**Reply by chapmandu** [reply]
Hey mate.. That's great news!

Which version did you upgrade from.. I'd like to find out which particular change/s may have made the difference.

**Reply by edthetechie** [reply]
It was on 2.0.2 for this app, I wasn't able to update it before because it's fairly old code that used a load of legacy wildcard routing and every time I upgraded it just errored out. The other good thing I noticed today was that the ACF server exception.log was filling up with large numbers of errors like this:
```
"Error","scheduler-3","09/07/22","16:42:43",,"Unable to duplicate a ColdFusion Component.coldfusion.runtime.UndefinedVariableException: Variable APPLICATION is undefined. The specific sequence of files included or processed is: '', line: 156 "
coldfusion.runtime.ObjectDuplicator$UnableToDuplicateCFCException: Unable to duplicate a ColdFusion Component.
```
By "large" I mean every few seconds but they were silent errors. I posted about it in the old Google Group. Anyway, those errors have gone now as well. I'm not sure if it's related to our performance improvements but I'm as happy as a pig in poo right now. What a fabulous start to my week  :-)

**Reply by neokoenig** [reply]
FYI one of the main reasons for the performance boost would probably have been the reversion to the plugin framework which was pretty unperformant with early 2.0/1 releases.

**Reply by chapmandu** [reply]
Ah yes.. @neokoenig hit the mail on the head. That plug-in code was indeed the culprit. It's a pity @edthetechie you put up with it for so long..

**Reply by edthetechie** [reply]
I thought it was my crappy code. I built this complex form when I was first learning CFWheels and it was really slow from the get-go so I figured I'd done something dumb but could never quite work out what it was. Never mind, all's well that ends well :-)

**Comment by bpamiri** [comment]
Ed,

You may want to give v2.4 another try as for the wildcard routes goes. That bug was fixed in v2.3 which had basically disabled the wildcard routes. With it working you may be able to update your v2.0.2 project to v2.4 without much effort.
 
--Peter Amiri
  ***@***.***

**Reply by edthetechie** [reply]
Thanks Peter, I actually did that on Friday and it worked beautifully
